sgv2-graphqlapi/src/main/java/io/stargate/sgv2/graphql/schema/cqlfirst/dml/FieldTypeCache.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.sgv2.graphql.schema.cqlfirst.dml;
+
+import graphql.Scalars;
+import graphql.schema.GraphQLScalarType;
+import io.stargate.proto.QueryOuterClass.TypeSpec;
+import io.stargate.sgv2.graphql.schema.scalars.CqlScalar;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Caches a category of GraphQL field types, corresponding to table columns or UDT fields.
+ *
+ * <p>There are different categories, each implemented by a subclass. Note that this cache does not
+ * contain table types, they are handled as top-level entities in {@link DmlSchemaBuilder}.
+ *
+ * @param <GraphqlT> the returned GraphQL type.
+ */
+abstract class FieldTypeCache<GraphqlT> {
+
+  protected final NameMapping nameMapping;
+  private final Map<TypeSpec, GraphqlT> types = new HashMap<>();
```

Comment:
this is indented to be non-thread safe right?

Reply:
Yes. These caches only exist for the duration of a `DmlSchemaBuilder.build()` call, and that operation is confined to a single thread.